sdk/table/azure-table/azure/table/_table_client.py

```diff
@@ -0,0 +1,575 @@
+                request_id_parameter=request_id_parameter,
+                **kwargs)
+            return response
+        except HttpResponseError as error:
+            process_storage_error(error)
+
+    @distributed_trace
+    def delete_entity(
+            self,
+            partition_key,
+            row_key,
+            etag=None,
+            match_condition=None,
+            **kwargs
+    ):
+        # type: (...) -> None
+        """Deletes the specified entity in a table.
+
+        :param match_condition: MatchCondition
+        :type match_condition: ~azure.core.MatchConditions
+        :param etag: Etag of the entity
+        :type etag: str
+        :param partition_key: The partition key of the entity.
+        :type partition_key: str
+        :param row_key: The row key of the entity.
+        :type row_key: str
+        :keyword callable cls: A custom type or function that will be passed the direct response
+        :return: None, or the result of cls(response)
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        if_match, if_not_match = _get_match_headers(kwargs=dict(kwargs, etag=etag, match_condition=match_condition),
+                                                    etag_param='etag', match_param='match_condition')
+
+        try:
+            self._client.table.delete_entity(
+                table=self.table_name,
+                partition_key=partition_key,
+                row_key=row_key,
+                if_match=if_match or if_not_match or '*',
+                **kwargs)
+        except ResourceNotFoundError:
+            raise ResourceNotFoundError
+
+    @distributed_trace
+    def create_entity(
+            self,
+            headers=None,
+            table_entity_properties=None,
+            query_options=None,
+            response_hook=None,  # pylint:disable=W0613
+            **kwargs
+    ):
+        # type: (...) -> Dict[str, object]
+        """Insert entity in a table.
+
+        :param response_hook:
+        :type response_hook:
+        :param headers: Headers for service request
+        :type headers: HttpResponse Headers
+        :param table_entity_properties: The properties for the table entity.
+        :type table_entity_properties: dict[str, object]
+        :param query_options: Parameter group.
+        :type query_options: ~azure.table.models.QueryOptions
+        :keyword callable cls: A custom type or function that will be passed the direct response
+        :return: dict mapping str to object, or the result of cls(response)
+        :rtype: dict[str, object] or None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        if table_entity_properties:
+
+            if "PartitionKey" in table_entity_properties and "RowKey" in table_entity_properties:
+                table_entity_properties = _add_entity_properties(table_entity_properties)
+            else:
+                raise ValueError
```

Comment:
this should probably include a message saying something about partition key and row key being required